### PR TITLE
feat: DEV-1595: Implement sampling-based assignments on SDK

### PR DESCRIPTION
**Pull request by KonstantinKorotaev**

### CODE REVIEW

File: label_studio_sdk/project.py [hunk 1]
```diff
@@ -1361,3 +1365,122 @@ def connect_azure_export_storage(
...
+        """
+        final_results = []
+        # Get tasks to assign
+        tasks = self.get_tasks(view_id=view_id, only_ids=True)
+        # Choice fraction of tasks
+        if fraction != 1.0:
+            k = int(len(tasks) * fraction)
+            tasks = sample(tasks, k)
+        # Check how many tasks for each user
+        n_tasks = max(int(len(tasks) / len(users)), 1)
```
Comment (makseq) [inline label_studio_sdk/project.py:1405]: Please do this at the beginning to avoid zero division: 
```
assert len(users) > 0, 'Users are empty'
```

Comment (makseq) [reply label_studio_sdk/project.py:1405]: Also we need to check tasks 
```
if len(tasks) == 0:
  logger.warning('No tasks found')
  return None
```

File: label_studio_sdk/project.py [hunk 2]
```diff
@@ -1361,3 +1365,122 @@ def connect_azure_export_storage(
...
+        # Assign each user tasks
+        for user in users:
+            if n_tasks > len(tasks):
+                n_tasks = len(tasks)
+            elif n_tasks + 1 == len(tasks):
```
Comment (makseq) [inline label_studio_sdk/project.py:1410]: Seems it is not very clear logic here, could you please write why do we need `n_tasks = n_tasks + 1` for all users?

**Review by makseq — CHANGES_REQUESTED** [review]

File: label_studio_sdk/project.py (hunk 2 above)
Comment (KonstantinKorotaev) [reply label_studio_sdk/project.py:1410]: These logic covers rounding for tasks chunks (see line 1407).
Added comments to clarify why it's used

File: label_studio_sdk/project.py (hunk 1 above)
Comment (KonstantinKorotaev) [reply label_studio_sdk/project.py:1405]: Fixed

File: label_studio_sdk/project.py [hunk 3]
```diff
@@ -1361,3 +1384,126 @@ def connect_azure_export_storage(
         }
         response = self.make_request('POST', '/api/storages/export/azure', json=payload)
         return response.json()
+
+    def assign_by_sampling(
```
Comment (niklub) [inline label_studio_sdk/project.py:1388]: I feel like this method should be "_internal"

File: label_studio_sdk/project.py [hunk 4]
```diff
@@ -1361,3 +1384,126 @@ def connect_azure_export_storage(
...
+        for user in users:
+            # check if last chunk of tasks is less than average chunk
+            if n_tasks > len(tasks):
+                n_tasks = len(tasks)
+            # check if last chunk of tasks is more than average chunk + 1 (covers rounding issue in line 1407)
+            elif n_tasks + 1 == len(tasks):
+                n_tasks = n_tasks + 1
+            if method == AssignmentSamplingMethod.RANDOM:
+                sample_tasks = sample(tasks, n_tasks)
+            else:
```
Comment (niklub) [inline label_studio_sdk/project.py:1437]: raise for the wrong method

**Review by niklub — CHANGES_REQUESTED** [review]

File: label_studio_sdk/project.py (hunk 4 above)
Comment (KonstantinKorotaev) [reply label_studio_sdk/project.py:1437]: Fixed

File: label_studio_sdk/project.py (hunk 3 above)
Comment (KonstantinKorotaev) [reply label_studio_sdk/project.py]: Fixed

**Pull request merged**